### PR TITLE
Add Join & Leave API endpoints with tests

### PR DESCRIPTION
**Pull request by Ramko9999**
* Added some utility DB functions in `cli.py`
* Setup testing with pytest
* Added endpoints for players to join and leave a session

**Create a .env file if you haven't already done so and add the following:**
```
GITHUB_ACCESS_TOKEN=SOME_ACCESS_TOKEN
DB_URI=postgres://postgres:gitgame_password@host.docker.internal:5433/gitgame_db
TEST_DB_URI=postgres://postgres:gitgame_password@localhost:5433/gitgame_db_test
```

If pytest isn't picking up imports from the `api` package inside api-server, be sure to run `pip install -e .` within the api-server directory.

### CODE REVIEW

File: api-server/api/routes/session.py [hunk 1]
```diff
@@ -17,3 +30,98 @@ async def make_session():
...
+)
+async def join(session_id: str, join_body: JoinRequestBody):
+    # todo(ramko9999): select for update may prevent insertions of rows which reference session_id via Fkey. Consider passing in no_key
+    # or using an other table just for locking lobby related activities such as joining, leaving or scoring the game.
+    session = await Session.filter(id=session_id).select_for_update().first()
+    if not session:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND, f"Session {session_id} not found"
+        )
+    player_id = session_id + "-" + join_body.username
```
Comment (rohanshiva) [inline api-server/api/routes/session.py]:
```suggestion
    player_id = f"{session_id}-{join_body.username}"
```

File: api-server/api/routes/session.py [hunk 2]
```diff
@@ -17,3 +30,98 @@ async def make_session():
...
+    async with in_transaction():
```
Comment (rohanshiva) [inline api-server/api/routes/session.py:65]: is `in_transaction` like tortoise db way of executing queries in transaction?

Comment (Ramko9999) [reply api-server/api/routes/session.py:65]: yes

File: api-server/api/routes/session.py [hunk 3]
```diff
@@ -17,3 +30,98 @@ async def make_session():
...
+        player = await Player.get_or_none(id=player_id)
+        if player and player.connection_state == Player.ConnectionState.CONNECTED:
+            raise HTTPException(
+                status.HTTP_422_UNPROCESSABLE_ENTITY,
+                f"Player {join_body.username} has already connected to session {session_id}",
+            )
+
+        if player:
+            player.connection_state = Player.ConnectionState.CONNECTED
```
Comment (rohanshiva) [inline api-server/api/routes/session.py:74]: When would a player be in this state?

File: api-server/api/routes/session.py [hunk 4]
```diff
@@ -17,3 +30,98 @@ async def make_session():
...
+    "/leave/{session_id}", status_code=status.HTTP_202_ACCEPTED, response_model=Lobby
+)
+async def leave(session_id: str, leave_body: LeaveRequestBody):
+    session = await Session.filter(id=session_id).select_for_update().first()
+    if not session:
+        # todo(Ramko9999): make this session check a dependency
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND, f"Session {session_id} not found"
+        )
+    player_id = session_id + "-" + leave_body.username
```
Comment (rohanshiva) [inline api-server/api/routes/session.py]:
```suggestion
    player_id = f"{session_id}-{leave_body.username}"
```

File: api-server/api/routes/session.py [hunk 5]
```diff
@@ -17,3 +30,98 @@ async def make_session():
...
+    async with in_transaction():
+        player = await Player.get_or_none(id=player_id)
+        if not player or player.connection_state == Player.ConnectionState.DISCONNECTED:
+            raise HTTPException(
+                status.HTTP_422_UNPROCESSABLE_ENTITY,
```
Comment (rohanshiva) [inline api-server/api/routes/session.py:109]: does it make sense to return 422 here?

Comment (rohanshiva) [reply api-server/api/routes/session.py:109]: nvm this works

File: api-server/test/conftest.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+from httpx import AsyncClient
+from tortoise import Tortoise
+from config import TEST_DB_URI
+from api.app import app
+from api.models import Session, Player
+
+MODELS = [Session, Player]
+
+
+@pytest.fixture(scope="session")
```
Comment (rohanshiva) [inline api-server/test/conftest.py:11]: what does this scope mean?

Comment (Ramko9999) [reply api-server/test/conftest.py:11]: I believe this fixture is used when running async tests with `anyio` as seen in `session_test.py` where I used the anyio decorator on the tests [Testing with AnyIO](https://anyio.readthedocs.io/en/stable/testing.html#creating-asynchronous-tests). I believe `scope=session` represents that the fixture will be run once in a test session and its output will be reused for the duration of the test session.

**Pull request merged**